common/mixins/modal.js

```diff
@@ -35,11 +35,25 @@ export default {
      * @param {object} el - optional - ref of dom element to trap focus on.
      *  will default to the root node of the vue component
      */
-    async focusFirstElement (el) {
+    async focusFirstElement (el = this.$el) {
       const elToFocus = await this.getFirstFocusableElement(el);
       elToFocus?.focus({ preventScroll: true });
     },
 
+    async focusElementById (elementId) {
+      await this.$nextTick();
+      const result = this.$el.querySelector(elementId);
+      if (result) {
+        result.focus();
+        return;
+      }
+
+      // eslint-disable-next-line no-console
+      console.warn('Could not find the element specified in dt-modal prop "initialFocusElement". ' +
+          'Defaulting to focusing the first element.');
+      await this.focusFirstElement();
+    },
+
```

Comment:
Code suggestion:

```
async focusElementById (elementId) {
      await this.$nextTick();

      try {
        const result = this.$el.querySelector(elementId);

        if (result) {
          result.focus();
          return;
        }
      } catch (error) {
        console.error(`Error occurred while executing querySelector: ${error}`);
      }

      console.warn('Could not find the element specified in dt-modal prop "initialFocusElement". ' +
          'Defaulting to focusing the first element.');
      await this.focusFirstElement();
    },
```

Reply:
I modify the `initialFocusElement` prop in the controls then the component crashes.... Just in case maybe a try/catch and a console.error handle this properly

Reply:
I don't think it's necessary to handle that error, it's simplier to do this.

`const result = this.$el?.querySelector(elementId);`

That way if for some reason the $el doesn't exist it will be undefined and focus the first element by default.

What was the case you were thinking that error could be triggered?

Reply:
> I modify the `initialFocusElement` prop with a "#" in the controls to trigger the `else if (this.initialFocusElement.startsWith('#')) {` then the component crashes.... Just in case maybe a try/catch and a console.error handle this properly

Did you set the property to just #button or "#button" it should be wrapped by quotes in order to properly work for storybook.

Reply:
Ye, right, that can be also a possible solution , it is working well with that.

Y set the property to only "#" and its crash, but now its works (cos the optional channing)